src/main/java/com/movierecommender/project/MovieController.java

```diff
@@ -1,67 +1,57 @@
-import java.net.http.HttpRequest;
-import java.net.http.HttpResponse;
-import java.io.IOException;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
 
 @Controller
 public class MovieController {
 
-    @GetMapping("/movie")
-    public String getMovie() {
-        String url = "https://api.themoviedb.org/3/movie/24?language=en-US";
+    private final MovieService movieService;
 
-        HttpClient client = HttpClient.newHttpClient();
-        HttpRequest request = HttpRequest.newBuilder()
-                .uri(URI.create(url))
-                .header("accept", "application/json")
-                .header("Authorization", "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI2NmZmYTM2NjgyYTY5ZDM1OWQ1MjQ4OTFkNDQ1OGI2NiIsIm5iZiI6MTcyOTYyNDQ5MS43MjE2MTgsInN1YiI6IjY3MTdmODBjNmU0MjEwNzgwZjc4NzRlYyIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.ubnKxHe4mfL51OE8NhOz31gacsvQU0wyh1Ja6vfp4D0") // Replace with your actual token
-                .GET()
-                .build();
+    @Autowired
+    public MovieController(MovieService movieService) {
+        this.movieService = movieService;
+    }
 
-        try {
-            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
-            if (response.statusCode() == 200) {
-                return response.body();
-            } else {
-                return "Error: " + response.statusCode();
-            }
-        } catch (IOException | InterruptedException e) {
-            e.printStackTrace();
-            return "Error occurred while fetching movie data.";
-        }
+    @GetMapping("/movie")
+    public String getMovie() {
+        return movieService.getMovie();
     }
 
-    //Added mapping for html pages
+    // HTML page mappings
     @GetMapping("/")
-    public String Index()
-    {
+    public String index() {
         return "Index";
     }
+
     @GetMapping("/login")
-    public String Login()
-    {
+    public String showSigninForm(@ModelAttribute("signin") Signin signin) {
         return "login";
     }
+
     @GetMapping("/signup")
-    public String Signup()
-    {
+    public String signup() {
         return "signup";
     }
-    @GetMapping("/Browse")
-    public String Browse()
-    {
-        return "browse";
-    }
-    @GetMapping("/Profile")
-    public String Profile()
-    {
+
+    @GetMapping("/profile")
+    public String profile() {
         return "profile";
     }
-}
+
+    @PostMapping("/signin")
+    public String processSignin(@Valid @ModelAttribute("signin") Signin signin,
+                                BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
+            return "login"; // If errors are present, return to signin page with error messages
+        }
+        // If no errors, process the login logic (e.g., authentication)
+        return "redirect:/home"; // Redirect to home or another page upon successful login
+    }
```

Comment:
This method was added to handle the form submission with validation by using @Valid and BindingResult, this method made sure that if there are any validation errors the user is returned to the login page with error messages shown to the user.